test/unit/deployment/test_cli.py

```diff
@@ -0,0 +1,261 @@
+def container_file(tmp_path):
+    file = tmp_path / "container.tgz"
+    file.touch()
+    return file
+
+
+class OptionMapper:
+    """
+    This class enables to align the corresponding names of
+    * CLI option
+    * kwarg in call to LanguageContainerDeployer()
+    * sample value to be verified in a test case
+    * environment variable (optional)
+    """
+    def __init__(
+            self,
+            api: str,
+            value: any = None,
+            cli: str = None,
+            env: str = None,
+            prompt: str = None,
+    ):
+        self.api_kwarg = api
+        self.value = f"om_value_{api}" if value is None else value
+        self.cli = cli or "--" + api.replace("_", "-")
+        self.env = env
+        self.prompt = prompt
+
+    @classmethod
+    def from_secret_param(
+            cls,
+            param: SecretParams,
+            prompt: str = None,
+    ) -> "OptionMapper":
+        name = param.name
+        return cls(
+            api=name.lower(),
+            cli=f"--{param.value}",
+            env=name,
+            prompt=prompt,
+        )
+
+def test_parameter_formatters_1param():
+    cmd = click.Command('a_command')
+    ctx = click.Context(cmd)
+    opt = click.Option(['--version'])
+    formatters = _ParameterFormatters()
+    formatters.set_formatter(CustomizableParameters.container_url, 'http://my_server/{version}/my_stuff')
+    formatters.set_formatter(CustomizableParameters.container_name, 'downloaded')
+    formatters(ctx, opt, '1.3.2')
+    assert ctx.params[CustomizableParameters.container_url.name] == 'http://my_server/1.3.2/my_stuff'
+    assert ctx.params[CustomizableParameters.container_name.name] == 'downloaded'
+
+
+def test_parameter_formatters_2params():
+    cmd = click.Command('a_command')
+    ctx = click.Context(cmd)
+    opt1 = click.Option(['--version'])
+    opt2 = click.Option(['--user'])
+    formatters = _ParameterFormatters()
+    formatters.set_formatter(CustomizableParameters.container_url, 'http://my_server/{version}/{user}/my_stuff')
+    formatters.set_formatter(CustomizableParameters.container_name, 'downloaded-{version}')
+    formatters(ctx, opt1, '1.3.2')
+    formatters(ctx, opt2, 'cezar')
+    assert ctx.params[CustomizableParameters.container_url.name] == 'http://my_server/1.3.2/cezar/my_stuff'
+    assert ctx.params[CustomizableParameters.container_name.name] == 'downloaded-1.3.2'
+
+
+def test_deployer_cli_with_missing_container_option():
+    result = click.testing.CliRunner().invoke(
+        language_container_deployer_main,
+        ["--language-alias", "PYTHON3_PEC_TESTS_CLI",
+         "--bucketfs-user", "bfs-username",
+         "--bucketfs-password", "bfs-password",
+         "--dsn", "host:port",
+         "--db-user", "db-username",
+         "--db-pass", "db-password",
+         ])
+    assert result.exit_code == 1 and \
+        isinstance(result.exception, ValueError) and \
+        "Incomplete parameter list." in str(result.exception)
```

Comment:
Could you check that the error is caused by the missing container option?
In theory, it could be another error....

Reply:
This test was copied. I added the requested check. Please see latest push.